src/main/java/org/assertj/core/api/junit/jupiter/SoftAssertionsExtension.java

```diff
@@ -88,41 +100,45 @@
   @Override
   public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
     // Abort if parameter type is unsupported.
-    if (isUnsupportedParameterType(parameterContext.getParameter())) return false;
-
+    Class<? extends SoftAssertionsProvider> type = getSupportedParameterType(parameterContext.getParameter());
+    if (type == null) {
+    	return false;
+    }
+    
     Executable executable = parameterContext.getDeclaringExecutable();
     // @Testable is used as a meta-annotation on @Test, @TestFactory, @TestTemplate, etc.
     boolean isTestableMethod = executable instanceof Method && isAnnotated(executable, Testable.class);
     if (!isTestableMethod) {
-      throw new ParameterResolutionException(format("Configuration error: cannot resolve SoftAssertions or BDDSoftAssertions for [%s]. Only test methods are supported.",
+      throw new ParameterResolutionException(format("Configuration error: cannot resolve SoftAssertionsProvider instances for [%s]. Only test methods are supported.",
                                                     executable));
     }
+    if (Modifier.isAbstract(type.getModifiers())) {
+        throw new ParameterResolutionException(format("Configuration error: cannot resolve abstract SoftAssertionsProvider instances [%s].",
```

Comment:
the error is not super clear, I would phrase it more like:
```
the resolved SoftAssertionsProvider instance is actually an abstract class (%s) and thus can't be instantiated. 
```

Reply:
I replaced "instance" with "type", as we don't actually have the instance (indeed we can't).